### ISSUE TITLE
Suggested library: busted

### ISSUE THREAD

**Issue by italomaia** [dibyendumajumdar/Suravi#1]
Testing library for lua: https://olivinelabs.com/busted/

**Comment by dibyendumajumdar** [comment]
thank you - will review. agree that best to have a standard unit test library included

**Comment by dibyendumajumdar** [comment]
btw Torch comes with a test framework too. See https://github.com/dibyendumajumdar/ravi-torch7/blob/master/doc/tester.md

**Comment by italomaia** [comment]
Well, torch do seem to overlap with a lot of internal tools. Probably not the time to wonder if that is a good thing, but, wasn't there a warning that torch is being "replaced" by pyTorch? Should this distro rely so much in a soon-to-be defunct project? Not questioning torch greatness, of course. Just ... wondering.

**Comment by dibyendumajumdar** [comment]
Yes Torch is sadly now under maintenance. But that is okay as I expect to maintain and support my fork of Torch long term. Re the tools that overlap - that in itself is not an issue - I guess I have to review carefully whether the tool is used by enough projects. To be honest it may be that none of the libraries I am including use Busted (I haven't checked).

**Comment by italomaia** [comment]
If you plan to keep your torch fork, long-term, maybe, shifting from non-core "under maintenance" libraries would be the way to go. In the long-run that would be less code to maintain. Just a thought.

**Comment by italomaia** [comment]
Regarding busted usage, I've found that the libraries listed in the readme do not use busted. 

Some fellow suggested busted usage for luasocket (and that's about it): https://github.com/diegonehab/luasocket/issues/227

**Comment by dibyendumajumdar** [comment]
Although it would be nice to migrate stuff in torch it is less of an issue, I guess easier option is to add the most commonly used unit testing framework
